src/main/java/org/opensearch/security/auth/BackendRegistry.java

```diff
@@ -688,6 +697,19 @@ private boolean isBlocked(InetAddress address) {
         return false;
     }
 
+    public static boolean matchesHostPatterns(WildcardMatcher hostMatcher, InetAddress address) {
+        if (hostMatcher == null) {
+            return false;
+        }
+        if (address != null) {
+            final String ipAddress = address.getHostAddress();
+            final String hostname = address.getHostName();
```

Comment:
nit: this requires a reverse DNS record (PTR) record to be set for the IP address. This should be covered in documentation of this new config.  

https://docs.oracle.com/javase/8/docs/api/java/net/InetAddress.html#getHostName--

Reply:
I'm thinking about modifying this to take `hostResolverMode` into consideration as is done here: https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/securityconf/ConfigModelV7.java#L1227-L1252

Reply:
Added a check for `hostResolverMode`

Reply:
Thanks, LGTM!